src/frontends/pytorch/src/op/repeat_interleave.cpp

```diff
@@ -45,43 +49,56 @@ OutputVector translate_repeat_interleave(const NodeContext& context) {
         // repeats is Constant with more then 1 element
-        auto repeats = repeats_fw_node->cast_vector<int32_t>();
+        auto repeats = repeats_const->cast_vector<int32_t>();
         if (context.input_is_none(2)) {
             // case (repeats=tensor, dim=None)
             auto flat_shape = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {-1}));
             auto reshape = context.mark_node(std::make_shared<v1::Reshape>(input, flat_shape, false));
-            OutputVector all_indices = generate_indices_from_repeats_tensor(context, repeats);
-            auto concat = context.mark_node(std::make_shared<v0::Concat>(all_indices, 0));
-            result = std::make_shared<v8::Gather>(reshape, concat, const_0);
+            auto all_indices = generate_indices_from_repeats_tensor(context, repeats);
+            result = std::make_shared<v8::Gather>(reshape, all_indices, const_0);
         } else {
             // case (repeats=tensor, dim=number)
             auto dimension = context.get_input(2);
-            OutputVector all_indices = generate_indices_from_repeats_tensor(context, repeats);
-            auto concat = context.mark_node(std::make_shared<v0::Concat>(all_indices, 0));
-            result = std::make_shared<v8::Gather>(input, concat, dimension);
+            auto all_indices = generate_indices_from_repeats_tensor(context, repeats);
+            result = std::make_shared<v8::Gather>(input, all_indices, dimension);
         }
     } else {
-        // repeats is not Constant or single element constant
-        // Currently we support only case when repeats contains only one element. Otherwise next Reshape will fail.
-        auto repeats_input = context.mark_node(std::make_shared<v0::Convert>(context.get_input(1), element::i32));
-        repeats_input = context.mark_node(std::make_shared<v1::Reshape>(repeats_input, const_1_list, false));
-        auto repeats = context.mark_node(std::make_shared<v0::Concat>(OutputVector{repeats_input, const_1_list}, 0));
-        auto shape_perm = context.mark_node(v0::Constant::create(element::i32, Shape{2}, {1, 0}));
+        repeats = context.mark_node(std::make_shared<v0::Convert>(repeats, element::i32));
+        repeats = context.mark_node(std::make_shared<v1::Reshape>(repeats, const_neg_1, false));
         if (context.input_is_none(2)) {
             // case (repeats=number, dim=None)
-            auto flat_shape = context.mark_node(v0::Constant::create(element::i32, Shape{2}, {1, -1}));
-            auto reshape = context.mark_node(std::make_shared<v1::Reshape>(input, flat_shape, false));
-            auto tile = context.mark_node(std::make_shared<v0::Tile>(reshape, repeats));
-            auto transpose = context.mark_node(std::make_shared<v1::Transpose>(tile, shape_perm));
-            result = std::make_shared<v1::Reshape>(transpose, const_neg_1, false);
+            input = context.mark_node(std::make_shared<v1::Reshape>(input, const_neg_1, false));
```

Comment:
can you please add comments for each block of this logic.
Just split these steps into blocks and add comments

Reply:
Done